packages/app-accounts/src/modals/TransferWithType.tsx

```diff
@@ -6,62 +6,68 @@ import { SubmittableExtrinsic } from '@polkadot/api/promise/types';
+import assetRegistry from '../../../app-generic-asset/src/assetsRegistry';
 import translate from '../../../app-generic-asset/src/translate';
+import FormatBalance from '@polkadot/app-generic-asset/FormatBalance';
 
 interface Props extends I18nProps {
   className?: string;
   onClose: () => void;
   recipientId?: string;
   senderId?: string;
-  assets?: AssetsSubjectInfo;
 }
 
 interface Option {
   text: string;
   value: string;
 }
 
-function TransferWithType ({ assets, className, onClose, recipientId: propRecipientId, senderId: propSenderId, t }: Props): React.ReactElement<Props> {
+function TransferWithType ({ className, onClose, recipientId: propRecipientId, senderId: propSenderId, t }: Props): React.ReactElement<Props> {
   const { api } = useApi();
   const [assetId, setAssetId] = useState('0');
+  const [assetName, setAssetName] = useState<string>('');
   const [amount, setAmount] = useState<BN | undefined>(new BN(0));
   const [extrinsic, setExtrinsic] = useState<SubmittableExtrinsic | null>(null);
   const [hasAvailable, setHasAvailable] = useState(true);
-  const [hasBalance, setHasBalance] = useState(false);
+  const [hasBalance, setHasBalance] = useState(true);
   const [options, setOptions] = useState<Option[]>([]);
   const [recipientId, setRecipientId] = useState(propRecipientId || null);
   const [senderId, setSenderId] = useState(propSenderId || null);
-  let id = assetId;
-  let user = senderId;
-  const assetBalance = useCall<Balance>(api.query.genericAsset.freeBalance as any, [id, user]);
-
-  const retrivedAssetList = useCall<any>(
-    // @ts-ignore
-    api.rpc.genericAsset.registeredAssets as any,
-    []
-  );
-
-  const dropdownOptions = retrivedAssetList && retrivedAssetList.length > 0 && retrivedAssetList.map((asset: any) => {
-    const [assetId, assetInfo] = asset;
-
-    return {
-      text: u8aToString(assetInfo.symbol),
-      value: assetId.toString()
-    };
-  });
+  const [assetBalance, setAssetBalance] = useState<BN>(new BN(0));
 
+  // Query registered assets on first load
   useEffect((): void => {
-    setOptions(dropdownOptions || []);
-  }, [retrivedAssetList]);
+    // @ts-ignore
+    api.rpc.genericAsset.registeredAssets().then(
+      (assets: any) => {
+        const dropdownOptions: Option[] = assets && assets.length > 0 && assets.map((asset: any) => {
+          const [assetId, assetInfo] = asset;
+          return {
+            text: u8aToString(assetInfo.symbol),
+            value: assetId.toString()
+          };
+        });
+
+        setOptions(dropdownOptions);
+        setAssetId(dropdownOptions[0].value || "0");
+        setAssetName(dropdownOptions[0].text || "?");
+      });
+  }, []);
+
+  // Query balances on assetId or senderId change
+  useMemo((): void => {
+    // @ts-ignore
+    api.query.genericAsset.freeBalance(assetId, senderId || "").then(
+      (balance: any) => setAssetBalance((balance as Balance).toBn())
```

Comment:
Same for "any" here

Reply:
may be we can use (balance: Balance) here